### PR TITLE
Update how we seed agencies to get around id/name DB unique constraints

### LINKED ISSUE #375: Using find by ID in the db:seeds task does not support multiple envs
Using find by ID assumes that the IDs used in the `config/agencies.yml` matches the IDs of the objects found in every env/db (it seems these objects were exported from `int`, the one env where this task doesn't fail). 

Consider a fresh env where the database is newly created and agencies are loaded from the `config/agencies.yml` file. Note I modified the `db:seeds` task code as follows to highlight the issue:

```
Rails.application.config.agencies.each do |agency_id, values|
  agency = Agency.find_by(id: agency_id)
  if agency
    agency.update!(name: values['name'])
    puts "Updated #{values['name']}"
  else
    Agency.create!(name: values['name'])
    puts "Created #{values['name']}"
  end
rescue ActiveRecord::RecordNotUnique => error
  puts "Rescued #{values['name']}"
  #puts error.message
  #byebug
end
```

On first run you can see how the `db:seeds` task already overwrites values since it's using IDs from the yaml file:

```
% rake db:drop DISABLE_DATABASE_ENVIRONMENT_CHECK=1
/Users/jjg/Documents/sources/login/identity-dashboard/config/initializers/smtp.rb:1: warning: already initialized constant SMTP_SETTINGS
/Users/jjg/Documents/sources/login/identity-dashboard/config/initializers/smtp.rb:1: warning: previous definition of SMTP_SETTINGS was here
Dropped database 'identity-dashboard_production'
```

```
% bundle exec rake db:create db:migrate            
/Users/jjg/Documents/sources/login/identity-dashboard/config/initializers/smtp.rb:1: warning: already initialized constant SMTP_SETTINGS
/Users/jjg/Documents/sources/login/identity-dashboard/config/initializers/smtp.rb:1: warning: previous definition of SMTP_SETTINGS was here
Created database 'identity-dashboard_production'
== 20160509195741 AddUsersAndApplications: migrating ==========================
-- create_table(:users, {:id=>:integer})
   -> 0.0022s
-- add_index(:users, :uuid, {:unique=>true})
   -> 0.0017s
-- add_index(:users, :email, {:unique=>true})
   -> 0.0012s
-- create_table(:applications, {:id=>:integer})
   -> 0.0020s
-- add_index(:applications, :issuer, {:unique=>true})
   -> 0.0014s
== 20160509195741 AddUsersAndApplications: migrated (0.0087s) =================

...

== 20200611125630 AddEmailUuidWhereDeletedAtIndexesToUsers: migrating =========
-- add_index(:users, :email, {:name=>"index_users_on_email", :where=>"(deleted_at IS NULL)"})
   -> 0.0012s
-- add_index(:users, :uuid, {:name=>"index_users_on_uuid", :where=>"(deleted_at IS NULL)"})
   -> 0.0011s
== 20200611125630 AddEmailUuidWhereDeletedAtIndexesToUsers: migrated (0.0024s)
```

```
% bundle exec rake db:seed             
/Users/jjg/Documents/sources/login/identity-dashboard/config/initializers/smtp.rb:1: warning: already initialized constant SMTP_SETTINGS
/Users/jjg/Documents/sources/login/identity-dashboard/config/initializers/smtp.rb:1: warning: previous definition of SMTP_SETTINGS was here
Created Library of Congress
Created Judicial Branch
Created Department of Agriculture
Created Department of Commerce
Created Department of Defense
Created Department of Education
Created Department of Energy
Created Department of Justice
Created Department of Health and Human Services
Updated Department of Homeland Security
Created Department of Housing and Urban Development
Created Department of the Interior
Created Department of Labor
Created Department of State
Updated Department of Transportation
Created Department of the Treasury
Created Department of Veterans Affairs
Created Administrative Conference of the United States
Created AbilityOne Commission
Created Access Board
Created Advisory Council on Historic Preservation
Created African Development Foundation
Created American Battle Monuments Commission
Created Appalachian Regional Commission
Created Armed Forces Retirement Home
Created Barry Goldwater Scholarship and Excellence in Education Foundation
Created Central Intelligence Agency
Created Chemical Safety Board
Created Christopher Columbus Fellowship Foundation
Created Commission for the Preservation of America's Heritage Abroad
Created Commission of Fine Arts
Created Commission on Civil Rights
Created Commodity Futures Trading Commission
Created Consumer Financial Protection Bureau
Created Consumer Product Safety Commission
Created Corporation for National & Community Service
Created Corporation for Public Broadcasting
Created Council of Inspectors General on Integrity and Efficiency
Created Court Services and Offender Supervision
Created Defense Nuclear Facilities Safety Board
Created Delta Regional Authority
Created Denali Commission
Created Director of National Intelligence
Created District of Columbia Courts
Created Dwight D. Eisenhower Memorial Commission
Created Election Assistance Commission
Created Environmental Protection Agency
Created Equal Employment Opportunity Commission
Updated Executive Office of the President
Created Export/Import Bank of the U.S.
Created Farm Credit Administration
Created Farm Credit System Insurance Corporation
Created Federal Communications Commission
Created Federal Deposit Insurance Corporation
Created Federal Election Commission
Created Federal Energy Regulatory Commission
Created Federal Financial Institutions Examination Council
Created Federal Home Loan Bank System
Created Federal Home Loan Mortgage Corporation
Created Federal Housing Finance Agency
Created Federal Labor Relations Authority
Created Federal Maritime Commission
Created Federal Mediation and Conciliation Service
Created Federal Mine Safety and Health Review Commission
Created Federal National Mortgage Association
Created Federal Reserve Board of Governors
Created Federal Retirement Thrift Investment Board
Created Federal Trade Commission
Updated General Services Administration
Created Government Accountability Office
Created Government of the District of Columbia
Created Gulf Coast Ecosystem Restoration Council
Created Harry S. Truman Scholarship Foundation
Created Institute of American Indian and Alaska Native Culture and Arts Development
Created Institute of Museum and Library Services
Created Inter-American Foundation
Created International Development Finance Corporation
Created International Trade Commission
Created James Madison Memorial Fellowship Foundation
Created Japan-United States Friendship Commission
Created John F. Kennedy Center for the Performing Arts
Created Legal Services Corporation
Created Marine Mammal Commission
Created Merit Systems Protection Board
Created Millennium Challenge Corporation
Created Morris K. Udall and Stewart L. Udall Foundation
Created National Aeronautics and Space Administration
Created National Archives and Records Administration
Created National Capital Planning Commission
Created National Council on Disability
Created National Credit Union Administration
Created National Endowment for the Arts
Created National Endowment for the Humanities
Created National Gallery of Art
Updated National Geospatial-Intelligence Agency
Updated National Labor Relations Board
Created National Mediation Board
Created National Nanotechnology Coordination Office
Created National Nuclear Security Administration
Created National Science Foundation
Created National Security Agency
Created National Security Commission on Artificial Intelligence
Created National Transportation Safety Board
Created Networking Information Technology Research and Development
Created Northern Border Regional Commission
Created Nuclear Regulatory Commission
Created Nuclear Waste Technical Review Board
Created Occupational Safety & Health Review Commission
Created Office of Government Ethics
Created Office of Navajo and Hopi Indian Relocation
Updated Office of Personnel Management
Created Office of Special Counsel
Created Office of the Federal Coordinator for Alaska Natural Gas Transportation Projects
Created Patient-Centered Outcomes Research Trust Fund
Created Peace Corps
Created Pension Benefit Guaranty Corporation
Created Postal Regulatory Commission
Created Postal Service
Created Presidio Trust
Created Privacy and Civil Liberties Oversight Board
Created Public Buildings Reform Board
Created Public Company Accounting Oversight Board
Created Public Defender Service for the District of Columbia
Updated Railroad Retirement Board
Updated Securities and Exchange Commission
Created Securities Investor Protection Corporation
Created Selective Service System
Updated Small Business Administration
Created Smithsonian Institution
Created Social Security Administration
Created Social Security Advisory Board
Created State Justice Institute
Created Surface Transportation Board
Created Tennessee Valley Authority
Created Trade and Development Agency
Created U.S. Agency for Global Media
Updated U.S. Agency for International Development
Created U.S. Army Corps of Engineers
Created United Mine Workers of America Benefit Funds
Created United States Court of Appeals for Veterans Claims
Created United States Holocaust Memorial Museum
Created United States Institute of Peace
Created United States Interagency Council on Homelessness
Created Vietnam Education Foundation
Updated State, local governments or tribes
```

Note above how a number of objects are updated, not created on first run as expected. Subsequent runs of the `db:seeds` task result in output like this:

```
 % bundle exec rake db:seed
/Users/jjg/Documents/sources/login/identity-dashboard/config/initializers/smtp.rb:1: warning: already initialized constant SMTP_SETTINGS
/Users/jjg/Documents/sources/login/identity-dashboard/config/initializers/smtp.rb:1: warning: previous definition of SMTP_SETTINGS was here
Updated Library of Congress
Updated Judicial Branch
Updated Department of Agriculture
Updated Department of Commerce
Updated Department of Defense
Updated Department of Education
Rescued Department of Energy
Updated Department of Justice
Updated Department of Health and Human Services
Updated Department of Homeland Security
Rescued Department of Housing and Urban Development
Updated Department of the Interior
Rescued Department of Labor
Updated Department of State
Updated Department of Transportation
Rescued Department of the Treasury
Updated Department of Veterans Affairs
Updated Administrative Conference of the United States
Updated AbilityOne Commission
Updated Access Board
Updated Advisory Council on Historic Preservation
Updated African Development Foundation
Updated American Battle Monuments Commission
Rescued Appalachian Regional Commission
Updated Armed Forces Retirement Home
Updated Barry Goldwater Scholarship and Excellence in Education Foundation
Updated Central Intelligence Agency
Updated Chemical Safety Board
Updated Christopher Columbus Fellowship Foundation
Rescued Commission for the Preservation of America's Heritage Abroad
Updated Commission of Fine Arts
Updated Commission on Civil Rights
Updated Commodity Futures Trading Commission
Updated Consumer Financial Protection Bureau
Updated Consumer Product Safety Commission
Updated Corporation for National & Community Service
Updated Corporation for Public Broadcasting
Rescued Council of Inspectors General on Integrity and Efficiency
Updated Court Services and Offender Supervision
Updated Defense Nuclear Facilities Safety Board
Updated Delta Regional Authority
Updated Denali Commission
Updated Director of National Intelligence
Rescued District of Columbia Courts
Updated Dwight D. Eisenhower Memorial Commission
Updated Election Assistance Commission
Updated Environmental Protection Agency
Updated Equal Employment Opportunity Commission
Updated Executive Office of the President
Updated Export/Import Bank of the U.S.
Updated Farm Credit Administration
Updated Farm Credit System Insurance Corporation
Rescued Federal Communications Commission
Updated Federal Deposit Insurance Corporation
Updated Federal Election Commission
Updated Federal Energy Regulatory Commission
Updated Federal Financial Institutions Examination Council
Updated Federal Home Loan Bank System
Rescued Federal Home Loan Mortgage Corporation
Updated Federal Housing Finance Agency
Updated Federal Labor Relations Authority
Updated Federal Maritime Commission
Updated Federal Mediation and Conciliation Service
Updated Federal Mine Safety and Health Review Commission
Updated Federal National Mortgage Association
Updated Federal Reserve Board of Governors
Rescued Federal Retirement Thrift Investment Board
Updated Federal Trade Commission
Updated General Services Administration
Updated Government Accountability Office
Updated Government of the District of Columbia
Updated Gulf Coast Ecosystem Restoration Council
Updated Harry S. Truman Scholarship Foundation
Rescued Institute of American Indian and Alaska Native Culture and Arts Development
Updated Institute of Museum and Library Services
Updated Inter-American Foundation
Updated International Development Finance Corporation
Updated International Trade Commission
Updated James Madison Memorial Fellowship Foundation
Updated Japan-United States Friendship Commission
Updated John F. Kennedy Center for the Performing Arts
Rescued Legal Services Corporation
Updated Marine Mammal Commission
Updated Merit Systems Protection Board
Updated Millennium Challenge Corporation
Updated Morris K. Udall and Stewart L. Udall Foundation
Updated National Aeronautics and Space Administration
Rescued National Archives and Records Administration
Updated National Capital Planning Commission
Updated National Council on Disability
Updated National Credit Union Administration
Updated National Endowment for the Arts
Updated National Endowment for the Humanities
Updated National Gallery of Art
Updated National Geospatial-Intelligence Agency
Updated National Labor Relations Board
Rescued National Mediation Board
Updated National Nanotechnology Coordination Office
Updated National Nuclear Security Administration
Updated National Science Foundation
Updated National Security Agency
Updated National Security Commission on Artificial Intelligence
Rescued National Transportation Safety Board
Updated Networking Information Technology Research and Development
Updated Northern Border Regional Commission
Updated Nuclear Regulatory Commission
Updated Nuclear Waste Technical Review Board
Updated Occupational Safety & Health Review Commission
Updated Office of Government Ethics
Rescued Office of Navajo and Hopi Indian Relocation
Updated Office of Personnel Management
Updated Office of Special Counsel
Updated Office of the Federal Coordinator for Alaska Natural Gas Transportation Projects
Updated Patient-Centered Outcomes Research Trust Fund
Updated Peace Corps
Updated Pension Benefit Guaranty Corporation
Rescued Postal Regulatory Commission
Updated Postal Service
Updated Presidio Trust
Updated Privacy and Civil Liberties Oversight Board
Updated Public Buildings Reform Board
Updated Public Company Accounting Oversight Board
Updated Public Defender Service for the District of Columbia
Updated Railroad Retirement Board
Updated Securities and Exchange Commission
Rescued Securities Investor Protection Corporation
Updated Selective Service System
Updated Small Business Administration
Updated Smithsonian Institution
Created Social Security Administration
Created Social Security Advisory Board
Created State Justice Institute
Created Surface Transportation Board
Rescued Tennessee Valley Authority
Created Trade and Development Agency
Created U.S. Agency for Global Media
Updated U.S. Agency for International Development
Created U.S. Army Corps of Engineers
Created United Mine Workers of America Benefit Funds
Created United States Court of Appeals for Veterans Claims
Created United States Holocaust Memorial Museum
Rescued United States Institute of Peace
Created United States Interagency Council on Homelessness
Created Vietnam Education Foundation
Updated State, local governments or tribes
```

Note all the rescues. This can be resolved by ignoring the IDs from the yaml file which are irrelevant AFAICT and searching by agency name. I assume the ID was used as a measure to more gracefully update agency names however it does not work with multiple envs unless we also segregate the `agencies.yml` config file by environment. We can either do that or purge the DB agencies list with each deploy and update those values from the `agencies.yml`. I favor a `agencies.yml` without IDs that updates the DB with each deploy since I don't see a need for differing configs per env.

CC @achapm @zachmargolis

### PR DESCRIPTION
**Pull request by zachmargolis**
- Hopefully fixes #375

cc @jgrevich can you try this PR in your environment to make sure it works as expected?

### CODE REVIEW

**Review by mitchellhenke — APPROVED** [review]

**Comment by mitchellhenke** [comment]
I'm not sure if deleting will cause any issues with foreign key constraints though

**Comment by zachmargolis** [comment]
> I'm not sure if deleting will cause any issues with foreign key constraints though

ooof good point, `service_providers.agency_id` is a foreign key constraint to this table, so that would cause big problems for this

**Comment by zachmargolis** [comment]
> > I'm not sure if deleting will cause any issues with foreign key constraints though
> 
> ooof good point, `service_providers.agency_id` is a foreign key constraint to this table, so that would cause big problems for this

I retried this to add some renames to temporary names to get around deleting in 7fd26bf, I tried locally where I deleted an agency from the YAML, gave another agency its name, and re-ran the seed and it worked


This gives me an idea for a spec.... going to try to add that in a bit

**Comment by zachmargolis** [comment]
spec added in 9231b1771ccad2475d0419435b33ef7a898e9aff

File: spec/services/agency_seeder_spec.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    context 'when an existing service provider to an agency by id foreign key' do
+      let(:agency_configs) do
+        {
+          1 => { name: 'new name' },
+        }.with_indifferent_access
+      end
+
+      before do
+        Agency.delete_all
```
Comment (mitchellhenke) [inline spec/services/agency_seeder_spec.rb:62]: This can be deleted I think

**Review by mitchellhenke — APPROVED** [review]
One minor test thing, otherwise looks good 👍🏼

File: spec/services/agency_seeder_spec.rb (hunk 1 above)
Comment (zachmargolis) [reply spec/services/agency_seeder_spec.rb]: Good call, removed in 7359db88886395e3b3aaa6c81d7fcb6d47939484

**Pull request merged**